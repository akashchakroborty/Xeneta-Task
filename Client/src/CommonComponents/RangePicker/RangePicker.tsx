import React from 'react';
import moment from 'moment';
import { DatePicker } from 'antd';

const { RangePicker } = DatePicker;

export interface RangePickerProps {
  onChange: (dates: unknown, dateStrings: string[]) => void;
  dateFormat: string;
  startDate: string;
  endDate: string;
}

const RangePickerComponent: React.FC<RangePickerProps> = ({
  onChange,
  dateFormat,
  startDate,
  endDate,
}) => {
  return (
    <RangePicker
      onChange={onChange}
      value={[moment(startDate, dateFormat), moment(endDate, dateFormat)]}
      format={dateFormat}
    />
  );
};

export default RangePickerComponent;
