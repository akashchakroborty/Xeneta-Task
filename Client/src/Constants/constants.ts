export const SELECT_ORIGIN = 'Select Origin';
export const SELECT_DESTINATION = 'Select Destination';
export const GET_RATES = 'Get Rates';
export const NO_DATA_MESSAGE = 'Sorry, no data available for the selected ports.';
export const NO_DATA_MESSAGE_FOR_DATE_RANGE = 'Sorry, no data available for the selected date range.';
export const RATE_TYPE_MAPPER = new Map([
  ['Market Average', 'mean'],
  ['Market Low', 'low'],
  ['Market High', 'high'],
]);
export const DEFAULT_RATE_TYPE = 'mean';
export const EMPTY_STRING = '';
export const DEFAULT_PORTS = [];
export const DEFAULT_RATES = [];
export const DEFAULT_LOADING = false;
export const DEFAULT_GET_RATES_CLICKED = false;
export const DATE_FORMAT = 'YYYY-MM-DD';
