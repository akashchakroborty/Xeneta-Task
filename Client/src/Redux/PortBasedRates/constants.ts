export const SET_PORTS_SUCCESS = 'SET_PORTS_SUCCESS';
export const SET_PORTS_FAILURE = 'SET_PORTS_FAILURE';
export const GET_PORTS_LOADING = 'GET_PORTS_LOADING';
export const UPDATE_ORIGIN = 'UPDATE_ORIGIN';
export const UPDATE_DESTINATION = 'UPDATE_DESTINATION';
