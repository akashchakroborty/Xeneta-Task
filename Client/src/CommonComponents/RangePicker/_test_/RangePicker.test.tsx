import React from 'react';
import { shallow } from 'enzyme';
import RangePicker, { RangePickerProps } from '../RangePicker';
import { DATE_FORMAT, EMPTY_STRING } from '../../../Constants/constants';

describe('should test the RangePicker Component', () => {
  const mockProps: RangePickerProps = {
    onChange: jest.fn(),
    dateFormat: DATE_FORMAT,
    startDate: EMPTY_STRING,
    endDate: EMPTY_STRING,
  };
  test('should render the Component', () => {
    const component = shallow(<RangePicker {...mockProps} />);
    expect(component).toMatchSnapshot();
  });
});
