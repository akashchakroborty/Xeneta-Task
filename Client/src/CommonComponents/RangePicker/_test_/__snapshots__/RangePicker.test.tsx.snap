// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the RangePicker Component should render the Component 1`] = `
<RangePicker
  format="YYYY-MM-DD"
  onChange={[MockFunction]}
  value={
    Array [
      null,
      null,
    ]
  }
/>
`;
